### PR TITLE
obs-studio: Update to 30.0.2

### LINKED ISSUE #1001: Update obs-studio to 30.0.1
### Name

obs-studio

### Version

30.0.1

### Link to the release notes or announcement.

https://github.com/obsproject/obs-studio/releases/tag/30.0.1

### Other comments

_No response_

### PR DESCRIPTION
**Pull request by nelsontkq**
**Summary**

Resolves #1001

Release notes can be found [here](https://github.com/obsproject/obs-studio/releases/tag/30.0.2).

**Test Plan**

 1. Open OBS and test video, audio, and mic input.
 2. record to file and confirm it's working
 3. Add browser source and confirm that's working

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

**Comment by nelsontkq** [comment]
Using the browser plugin causes a segfault

**Comment by nelsontkq** [comment]
@Staudey this update appears to be in working order. If you want to give it a test / take over before I mark it ready for review feel free!

File: packages/o/obs-studio/package.yml [hunk 1]
```diff
@@ -28,23 +31,35 @@ builddeps  :
...
+    - pkgconfig(vpl)
     - pkgconfig(wayland-client)
     - pkgconfig(x264)
     - pkgconfig(xcomposite)
     - pkgconfig(xinerama)
     - pkgconfig(xkbcommon-x11)
+    - asio
     - cef-minimal-devel
+    - fdk-aac-devel
+    - libdatachannel-devel
```
Comment (nelsontkq) [inline packages/o/obs-studio/package.yml:46]: We can exclude this dep & package if we want to turn off the WebRTC feature `ENABLE_WEBRTC`

File: packages/o/obs-studio/package.yml [hunk 2]
```diff
@@ -28,23 +31,35 @@ builddeps  :
...
+    - pkgconfig(rnnoise)
     - pkgconfig(speexdsp)
     - pkgconfig(srt)
+    - pkgconfig(vpl)
     - pkgconfig(wayland-client)
     - pkgconfig(x264)
     - pkgconfig(xcomposite)
     - pkgconfig(xinerama)
     - pkgconfig(xkbcommon-x11)
+    - asio
```
Comment (Staudey) [inline packages/o/obs-studio/package.yml:43]: Already included as `pkgconfig(asio)`

**Comment by Staudey** [comment]
> @Staudey this update appears to be in working order. Since you're the maintainer, if you want to give it a test / take over before I mark it ready for review feel free!

I know the MAINTAINERS.md still mentions me, but I'm actually no longer contributing to Solus packages (apart from updating the Third Party repo). Feel free to replace my info in that file with yours.

I've added some comments to consider before someone does a real review of the PR.

File: packages/o/obs-studio/package.yml [hunk 3]
```diff
@@ -28,23 +31,35 @@ builddeps  :
...
     - cef-minimal-devel
+    - fdk-aac-devel
```
Comment (Staudey) [inline packages/o/obs-studio/package.yml:45]: Should be `pkgconfig(fdk-aac)`

File: packages/o/obs-studio/package.yml [hunk 4]
```diff
@@ -28,23 +31,35 @@ builddeps  :
...
+    - libdatachannel-devel
     - mbedtls-devel
+    - qrcodegencpp-devel
```
Comment (Staudey) [inline packages/o/obs-studio/package.yml:48]: Seems like that package has a .pc file and thus should be included with a `pkgconfig(...)`

**Comment by ReillyBrogan** [comment]
PRs always need their commits ordered in the order they will be built. It looks like you have OBS as the first commit even though it depends on later commits, you'll need to re-order it to be the last one. The reason why this is required is that our current build tooling is not smart enough to deduce the proper order from the packages themselves and so we push it to the build server in commit order. As it is this will fail the publish step.

**Comment by nelsontkq** [comment]
@ReillyBrogan re-ordered

**Review by davidjharder — APPROVED** [review]
Looks good, thanks

**Pull request merged**